### ISSUE TITLE
Truncate filenames longer than 255 characters

### ISSUE THREAD

**Issue by spencerhughes** [mlj/castget#40]
When using the filename template option, castget does not automatically truncate filenames longer than the Linux maximum filename length. This can cause issues, especially when using the %(title) option in the filename template, because many podcasts have very long titles. Either having it automatically truncate the names, or having an option that allows me to set the maximum file name length would solve the problem.

**Comment by mlj** [comment]
Agreed that some kind of truncation would make sense. We already strip some bad characters from filenames in `sanitise_filename()` in `src/filenames.c`, which would be a sensible place to add this.
